### PR TITLE
docs: Add how to create gpu workload cluster docs

### LINKED ISSUE #159: Document GPU support
GPU support is implemented via NVIDIA operator. 

How to do this should be added to the documentation.

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
This details how to launch a VM or BM GPU cluster using the Nvidia GPU-Operator

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #159

### CODE REVIEW

File: docs/src/gs/create-gpu-workload-cluster.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+      bootstrap:
+        configRef:
+          name: "${CLUSTER_NAME}-md"
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfigTemplate
+      infrastructureRef:
+        name: "${CLUSTER_NAME}-md"
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: OCIMachineTemplate
+      failureDomain: "2"
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-gpu-workload-cluster.md:322]: should we add a comment here?

File: docs/src/gs/create-gpu-workload-cluster.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+OCI_NODE_MACHINE_TYPE=VM.GPU3.1 \
+OCI_NODE_MACHINE_TYPE_OCPUS=6 \
+OCI_CONTROL_PLANE_MACHINE_TYPE_OCPUS=1 \
+OCI_CONTROL_PLANE_MACHINE_TYPE=VM.Standard3.Flex \
+CONTROL_PLANE_MACHINE_COUNT=1 \
+OCI_SHAPE_MEMORY_IN_GBS= \
+KUBERNETES_VERSION=v1.24.4 \
+clusterctl generate cluster <cluster-name> \
+--target-namespace default \
+--from cluster-template-failure-domain-spread.yaml | kubectl apply -f -
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-gpu-workload-cluster.md:72]: can we call the file name example-yaml or something? may be it is better to say copy the contents, create a file called cluster-template-gpu.yaml and then use that term ?

File: docs/src/gs/create-gpu-workload-cluster.md (hunk 1 above)
Comment (joekr) [reply docs/src/gs/create-gpu-workload-cluster.md:327]: fixed

File: docs/src/gs/create-gpu-workload-cluster.md (hunk 2 above)
Comment (joekr) [reply docs/src/gs/create-gpu-workload-cluster.md]: fixed

**Review by shyamradhakrishnan — APPROVED** [review]

**Pull request merged**